pym/bob/cmds/build.py

```diff
@@ -1429,20 +1446,30 @@ def doProject(argv, bobRoot):
     if args.preserve_env: extra.append('-E')
     if args.sandbox: extra.append('--sandbox')
 
-    package = packages.walkPackagePath(args.package)
 
-    # execute a bob dev with the extra arguments to build all executables.
+    backlog = []
+    for p in inputPackages:
+        for package in packages.queryPackagePath(p):
+            backlog.append(package)
+
+    # execute a bob dev/build with the extra arguments to build all executables.
     # This makes it possible for the plugin to collect them and generate some runTargets.
     if args.execute_prebuild:
         devArgs = extra.copy()
         if args.resume: devArgs.append('--resume')
         if args.execute_buildonly: devArgs.append('-b')
-        devArgs.append(args.package)
-        doDevelop(devArgs, bobRoot)
+        for p in backlog:
+            devArgs.append(p.getName())
+        if args.build:
+            doBuild(devArgs, bobRoot)
+        else:
+            doDevelop(devArgs, bobRoot)
+
+    print(">>", colorize("Generate project for", "32;1"))
+    for package in backlog:
+        print(colorize("   PROJECT   {} ({})".format(package.getName(), args.projectGenerator), "32"))
+    generator(backlog, generatorArguments, extra)
```

Comment:
Here must be a backwards compatible Plugin-API change. Old plugins will only get one package. If the user selected more than one package and the plugin is not updated we must show an error message.